vendor/github.com/prometheus/prometheus/tsdb/index/postings.go

```diff
@@ -108,11 +109,14 @@ func (p *MemPostings) SortedKeys() []labels.Label {
 	}
 	p.mtx.RUnlock()
 
-	slices.SortFunc(keys, func(a, b labels.Label) bool {
-		if a.Name != b.Name {
-			return a.Name < b.Name
+	slices.SortFunc(keys, func(a, b labels.Label) int {
+		nameCompare := strings.Compare(a.Name, b.Name)
```

Comment:
Would be great to open an upstream PR to not use `strings.Compare()` here. This should be something like:

```
if a.Name == b.Name {
  // ... compare using < > here ...
}
```